### DISCUSSION TITLE
cap-location doesn't seem to be working

### DISCUSSION THREAD

**Discussion by aghaynes** [quarto-dev/quarto-cli#11341] [Q&A] [answered]
### Description

I'm probably doing something wrong, but the cap-location options (including fig and tbl variants) don't seem to be doing anything... I initially tried it for typst, but it also doesnt seem to be working for html... In html, captions are at the top, in typst they're at the bottom... 

````
---
title: "Untitled"
format: html
editor: visual
keep-md: true
keep-typ: true
---

```{r}
#| tbl-cap: MTCARS
#| cap-location: top
library(gt)
mtcars[1:5, ] |> gt()
```

```{r}
#| tbl-cap: MTCARS
#| cap-location: bottom
library(gt)
mtcars[1:5, ] |> gt()
```
````
![image](https://github.com/user-attachments/assets/f487dd92-5a93-4282-90b1-523238c92417)

Am I doing something wrong?

I'm running quarto 1.5.45 through RStudio 2024.09.0 on windows 10.

**Comment by mcanouil** [comment]
`cap-location` is a global option which would also targets figures for instance.
Prefer `tbl-cap-location` unless you want figures and tables to follow the option.

I was going to say that I don't think the option(s) was supposed to work at the code cell level, but it is listed in the documentation: https://quarto.org/docs/reference/cells/cells-knitr.html#page-columns

Adding a label makes the code cell option work.

<table>
<tr><th>Input</th><th>Output</th></tr>
<tr><td>

````qmd
---
title: "Untitled"
format: html
---

```{r}
#| tbl-cap: MTCARS
#| tbl-cap-location: top
library(gt)
mtcars[1:5, ] |> gt()
```

```{r}
#| tbl-cap: MTCARS
#| tbl-cap-location: bottom
library(gt)
mtcars[1:5, ] |> gt()
```

````

</td><td>

<img width="624" alt="image" src="https://github.com/user-attachments/assets/9241a2af-b1e3-4908-960f-09a375c9f390">

</td></tr>
<tr><td>

````qmd
---
title: "Untitled"
format: html
---

```{r}
#| label: tbl-one
#| tbl-cap: MTCARS
#| tbl-cap-location: top
library(gt)
mtcars[1:5, ] |> gt()
```

```{r}
#| label: tbl-two
#| tbl-cap: MTCARS
#| tbl-cap-location: bottom
library(gt)
mtcars[1:5, ] |> gt()
```

````

</td><td>

<img width="593" alt="image" src="https://github.com/user-attachments/assets/94a31b0c-c36f-4523-8c17-f529c19577c3">

</td></tr>
</table>

It seems related to a previous fix:
- https://github.com/quarto-dev/quarto-cli/issues/9734

But It seems the fix did not make it all the way in 1.5

**Reply by mcanouil** [reply, marked as answer]
Thanks @aghaynes for the clean report/reprex, I've upgraded this to a bug issue:

- #11342

**Reply by aghaynes** [reply]
Thanks @mcanouil. Importantly, the label has to include tbl for it to work... but doing so does indeed make it work in both html and typst

**Reply by mcanouil** [reply]
Yes, the prefix is important because it means cross-referenceable.
A "cross-reference float" is a special thing in Quarto.
